### ISSUE TITLE
Support iojs 3.0.0

### ISSUE THREAD

**Issue by jbergstroem** [ncb000gt/node.bcrypt.js#322]
Breaks while compiling -- tried upgrading to nan 2.0.0. There is some api changes forcing a refactor. If no one else has time I can have a look at this in a week.

**Comment by coderhaoxin** [comment]
waiting for this issue to be fixed? https://github.com/nodejs/nan/issues/389

**Comment by defunctzombie** [comment]
Good to know about that one. Certainly will give these projects some time to bake. I understand folks want to test and upgrade quickly but some of the API changes will take a refactor. Without a migration guide for NaN 1.x to 2.x it will be slower going.

**Comment by s-a** [comment]
I try to compile for electron :crying_cat_face: 
ref:https://github.com/s-a/iron-node/issues/44
```bash
git clone --depth=1 https://github.com/ncb000gt/node.bcrypt.js.git c:\git\electron-recompile\tmp\bcrypt@0.8.4
Cloning into 'c:\git\electron-recompile\tmp\bcrypt@0.8.4'...
cd c:\git\electron-recompile\tmp\bcrypt@0.8.4 && set HOME=electron-gyp && node-gyp rebuild --target=0.30.2 --arch=ia32 --dist-url=https://atom.io/download/atom-shell
gyp info it worked if it ends with ok
gyp info using node-gyp@2.0.2
gyp info using node@0.12.6 | win32 | ia32
gyp http GET https://atom.io/download/atom-shell/v0.30.2/node-v0.30.2.tar.gz
gyp http 200 https://atom.io/download/atom-shell/v0.30.2/node-v0.30.2.tar.gz
gyp http GET https://atom.io/download/atom-shell/v0.30.2/SHASUMS256.txt
gyp http GET https://atom.io/download/atom-shell/v0.30.2/node.lib
gyp http GET https://atom.io/download/atom-shell/v0.30.2/x64/node.lib
gyp http 200 https://atom.io/download/atom-shell/v0.30.2/SHASUMS256.txt
gyp http 200 https://atom.io/download/atom-shell/v0.30.2/x64/node.lib
gyp http 200 https://atom.io/download/atom-shell/v0.30.2/node.lib
gyp info spawn c:\python27\python.exe
gyp info spawn args [ 'C:\\Users\\Stephan\\AppData\\Roaming\\npm\\node_modules\\node-gyp\\gyp\\gyp_main.py',
gyp info spawn args   'binding.gyp',
gyp info spawn args   '-f',
gyp info spawn args   'msvs',
gyp info spawn args   '-G',
gyp info spawn args   'msvs_version=auto',
gyp info spawn args   '-I',
gyp info spawn args   'c:\\git\\electron-recompile\\tmp\\bcrypt@0.8.4\\build\\config.gypi',
gyp info spawn args   '-I',
gyp info spawn args   'C:\\Users\\Stephan\\AppData\\Roaming\\npm\\node_modules\\node-gyp\\addon.gypi',
gyp info spawn args   '-I',
gyp info spawn args   'c:\\git\\electron-recompile\\tmp\\bcrypt@0.8.4\\electron-gyp \\.node-gyp\\0.30.2\\common.gypi',
gyp info spawn args   '-Dlibrary=shared_library',
gyp info spawn args   '-Dvisibility=default',
gyp info spawn args   '-Dnode_root_dir=c:\\git\\electron-recompile\\tmp\\bcrypt@0.8.4\\electron-gyp \\.node-gyp\\0.30.2',
gyp info spawn args   '-Dnode_gyp_dir=C:\\Users\\Stephan\\AppData\\Roaming\\npm\\node_modules\\node-gyp',
gyp info spawn args   '-Dmodule_root_dir=c:\\git\\electron-recompile\\tmp\\bcrypt@0.8.4',
gyp info spawn args   '--depth=.',
gyp info spawn args   '--no-parallel',
gyp info spawn args   '--generator-output',
gyp info spawn args   'c:\\git\\electron-recompile\\tmp\\bcrypt@0.8.4\\build',
gyp info spawn args   '-Goutput_dir=.' ]
module.js:338

    throw err;

          ^

Error: Cannot find module 'nan'

    at Function.Module._resolveFilename (module.js:336:15)

    at Function.Module._load (module.js:278:25)

    at Module.require (module.js:365:17)

    at require (module.js:384:17)

    at [eval]:1:1

    at Object.exports.runInThisContext (vm.js:74:17)

    at Object.<anonymous> ([eval]-wrapper:6:22)

    at Module._compile (module.js:460:26)

    at evalScript (node.js:431:25)

    at startup (node.js:90:7)

gyp: Call to 'node -e "require('nan')"' returned exit status 1. while trying to load binding.gyp
gyp ERR! configure error
gyp ERR! stack Error: `gyp` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onCpExit (C:\Users\Stephan\AppData\Roaming\npm\node_modules\node-gyp\lib\configure.js:357:16)
gyp ERR! stack     at ChildProcess.emit (events.js:110:17)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (child_process.js:1074:12)
gyp ERR! System Windows_NT 6.3.9600
gyp ERR! command "node" "C:\\Users\\Stephan\\AppData\\Roaming\\npm\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild" "--target=0.30.2" "--arch=ia32" "--dist-url=https://atom.io/download/atom-shell"
gyp ERR! cwd c:\git\electron-recompile\tmp\bcrypt@0.8.4
gyp ERR! node -v v0.12.6
gyp ERR! node-gyp -v v2.0.2
gyp ERR! not ok
Failed to compile : bcrypt

```

**Comment by defunctzombie** [comment]
What does this have to do with iojs 3? Please stay on issue topic.

**Comment by s-a** [comment]
It is so that electron is using iojs

**Comment by defunctzombie** [comment]
Not based on the log you posted. Anyhow. Let's focus on iojs here and not
other issues as it will just add noise. We know it doesn't build yet.

**Comment by s-a** [comment]
It is node: "2.3.1" and  v8: "4.3.61.21"

**Comment by nicolasgramlich** [comment]
They just fixed it: https://github.com/nodejs/nan/issues/389#issuecomment-129479306 the commit he referenced is part of the `2.0.5` tag: https://github.com/nodejs/nan/commit/98d38c182e30dd9dae2f8568996fb334e0ddcefc

**Comment by jbergstroem** [comment]
@defunctzombie [nan async pi](https://github.com/nodejs/nan/tree/master/examples/async_pi_estimate) works as a pretty good example for porting/implementing nan 2.0.0+

**Comment by defunctzombie** [comment]
PR here: https://github.com/ncb000gt/node.bcrypt.js/pull/324

Everyone who is interested please test this out and report issues.

**Comment by defunctzombie** [comment]
Released in 0.8.5
